### ISSUE TITLE
Latest-docs build on github pages is broken

### ISSUE THREAD

**Issue by pp-mo** [SciTools/iris#2945]
According to README, should be at https://scitools-docs.github.io/iris/master/index.html

Currently it's not there

**Comment by pelson** [comment]
Example build that removes all build files:

https://travis-ci.org/SciTools/iris/jobs/341513348

Pertinent stderrors:

```
 from /home/travis/build/SciTools/iris/docs/iris/src/examples/General/polar_stereo.py:
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/fileformats/__init__.py", line 70, in _load_grib
    from iris_grib import load_cubes
ModuleNotFoundError: No module named 'iris_grib'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/matplotlib/sphinxext/plot_directive.py", line 517, in run_code
    six.exec_(code, ns)
  File "<string>", line 28, in <module>
  File "<string>", line 19, in main
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/__init__.py", line 373, in load_cube
    cubes = _load_collection(uris, constraints, callback)
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/__init__.py", line 313, in _load_collection
    result = iris.cube._CubeFilterCollection.from_cubes(cubes, constraints)
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/cube.py", line 143, in from_cubes
    for cube in cubes:
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/__init__.py", line 300, in _generate_cubes
    for cube in iris.io.load_files(part_names, callback, constraints):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/io/__init__.py", line 210, in load_files
    for cube in handling_format_spec.handler(fnames, callback):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/scitools_iris-2.0.0rc2.dev0-py3.6.egg/iris/fileformats/__init__.py", line 72, in _load_grib
    raise RuntimeError('Unable to load GRIB file - '
RuntimeError: Unable to load GRIB file - "iris_grib" package is not installed.

```


```

Exception occurred:
  File "/home/travis/build/SciTools/iris/docs/iris/src/sphinxext/gen_gallery.py", line 196, in gen_gallery
    for key in app.builder.status_iterator(thumbnails,
AttributeError: 'StandaloneHTMLBuilder' object has no attribute 'status_iterator'
The full traceback has been saved in /tmp/sphinx-err-tkii_wjx.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make[1]: *** [html] Error 2
make: *** [html] Error 2
```

```
WARNING: autodoc: failed to import module 'iris.experimental.raster'; the following exception was raised:
No module named 'osgeo'
```

I wonder if just before https://travis-ci.org/SciTools/iris/jobs/341513348#L1119 we are getting an exit code and simply ignoring it?

**Comment by pelson** [comment]
I think we are paying the price for https://github.com/SciTools/iris/blob/master/.travis.yml#L130-L133:

```
  # "make html" produces an error when run on Travis that does not
  # affect any downstream functionality but causes the build to fail
  # spuriously. The echo-backtick workaround gets around this error,
  # which should be investigated further in the future.
```

Which was added as a comment >3 years ago.

**Comment by pelson** [comment]
I've just pushed a branch to upstream to see how it does with a potential fix. Branch is https://github.com/SciTools/iris/tree/travis_docs_build.  The build itself is at https://travis-ci.org/SciTools/iris/builds/341764492.

**Comment by pelson** [comment]
Fixed in #2958.
